### PR TITLE
Allow binding read expression in JDBC connectors

### PR DESCRIPTION
**Pull request by ebyhr**
Some databases need expressions to get the expected result via JDBC drivers.

### CODE REVIEW

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ReadFunction.java [hunk 1]
```diff
@@ -15,11 +15,17 @@
...
 import java.sql.SQLException;
+import java.util.Optional;
 
 public interface ReadFunction
 {
     Class<?> getJavaType();
 
+    default Optional<String> getBindExpression(String columnName)
+    {
+        return Optional.empty();
```
Comment (findepi) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ReadFunction.java:26]: Let's have default implementation which returns `columnName`.
Of course, this requires that we pass in a quoting function, but implementations will need anyway (any generic implementation; concrete impls can assume eg `"`-quoting)

**Review by findepi — COMMENTED** [review]
is it testable?

**Comment by hashhar** [comment]
It'll be nice to have a test (maybe using TestingH2JdbcClient?) since my reading of `QueryBuilder#prepareQuery` makes it look like we don't allow both expressions and constraints to exist at the same time and the expression you define in the read function won't be used when building the predicate (only used for building the projection).

**Comment by ebyhr** [comment]
> won't be used when building the predicate (only used for building the projection).
> the expression should be used always, in lieu of column name (column reference)

I'm not sure why read-expression should be used for predicates. Could you describe more details? 

We will use the expression with aggregation pushdown, but it would be evaluated after the aggregation in my understanding. e.g.
```
SELECT ${expression} AS `_pfgnrtd_0` FROM (SELECT max(`orderkey`) AS `_pfgnrtd_0` FROM `tpch`.`orders`) o
```

**Comment by findepi** [comment]
> I'm not sure why read-expression should be used for predicates. Could you describe more details?

Discussed offline.

**Comment by lhofhansl** [comment]
What happened to this?

**Pull request closed without merging**